### ISSUE TITLE
Shorthands don't work in table of contents after other language environment

### ISSUE THREAD

**Issue by wehro** [reutenauer/polyglossia#309]
```
\documentclass{article}
\usepackage{polyglossia}
\setmainlanguage[babelshorthands]{german}
\setotherlanguage{english}

\begin{document}
\tableofcontents
\section{A"/B}
\begin{english}
\section{English title}
\end{english}
\section{C"/D}
\end{document}
```
The table of contents says **C"/D** for the third section, but it should be **C/D** as `"/` is a shorthand.

**Comment by wehro** [comment]
```
\selectlanguage *[babelshorthands]{german}
\contentsline {section}{\numberline {1}A"/B}{1}% 
\selectlanguage *{english}
\contentsline {section}{\numberline {2}English title}{1}% 
\contentsline {section}{\numberline {3}C"/D}{1}%
```
The `toc` file shows that a `\selectlanguage` command is missing at the end of the English part.

**Comment by dohyunkim** [comment]
Then, language setting should be written into the .aux at the end of `english` environment. 
```diff
--- a/tex/polyglossia.sty
+++ b/tex/polyglossia.sty
@@ -1168,7 +1168,7 @@
 \newcommand*{\xpg@hook@setlanguage}{}

 \def\xpg@pop@language@i#1{%
-         \xpg@set@language@nonaux{#1}%
+         \xpg@set@language@aux{#1}%
          \xpg@hook@setlanguage%
          \let\emp@langname\@undefined}%

@@ -1177,7 +1177,7 @@
 {%
   \polyglossia@error@iflangnotloaded:n{#3}
   \expandafter\def\expandafter\xpg@pop@language\expandafter{\expandafter\xpg@pop@language@i\expandafter{\languagename}}
-      \aftergroup\xpg@pop@language%
+  \aftergroup\aftergroup\aftergroup\xpg@pop@language
   \tl_if_blank:nTF {#2}{}{%
      % If the optional argument sets a value for the key “variant”, copy it to xpg@langvariant
     \clist_map_inline:nn { #2 } {%
```
Duplication of `\aftergroup` seems to be necessary as the language environment of current master branch produces two group level. BTW, I'm not sure whether language option (`babelshorthands`) should also be written into .aux file.

**Comment by wehro** [comment]
Thank you, with this code change it works for me.
Could someone please apply the patch.

**Comment by jspitz** [comment]
Fixed with 68082354
